### PR TITLE
[EC-167] Fix error while running regular sync without all MPT nodes

### PR DESCRIPTION
**Pull request by ntallar**
## Description

In the scenario that:
* The client is started with fast sync enabled but it is canceled it before it completes
* The client is restarted with regular sync enabled.

currently it resulted in the client throwing the exception `Node not found, trie is inconsistent`.

## Fix
The exception is caused by trying to retrieve the MPT for the `BestBlockNumber` saved during fast sync, for which no MPT node is stored.
This was fixed by continuing with fast sync and logging this, even though regular sync was selected.

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala [hunk 1]
```diff
@@ -175,7 +176,7 @@ trait FastSync {
 
     private var blockChainOnlyPeers = Seq.empty[Peer]
 
-    private val syncStatePersistCancellable = scheduler.schedule(persistStateSnapshotInterval, persistStateSnapshotInterval, self, PersistSyncState)
+    private val syncStatePersistCancellable = scheduler.schedule(0.seconds, persistStateSnapshotInterval, self, PersistSyncState)
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala]: why 0 instead of named constant/configuration ?

File: src/main/scala/io/iohk/ethereum/utils/Config.scala [hunk 2]
```diff
@@ -94,7 +94,31 @@ object Config {
 
   }
 
-  object Sync {
+  trait SyncConfig {
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/utils/Config.scala]: 👍

File: src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala (hunk 1 above)
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala]: In this case this value shouldn't be configurable as while checking whether to allow starting regular sync or not, it is needed to [check whether fast sync had previously started](https://github.com/input-output-hk/etc-client/pull/287/files#diff-8c13c0426c84bb54f28f3662b267aa16R70).
For this it is checked whether any SyncState was saved, so right after `FastSync` starts, it's state should be saved to prevent it not being saved before `BestBlockNumber` is updated.

Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala]: ok, so maybe maned constant?

File: src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala [hunk 3]
```diff
@@ -65,8 +66,12 @@ class SyncController(
         case (true, false) =>
           startRegularSync()
         case (false, false) =>
-          fastSyncStateStorage.purge()
-          startRegularSync()
+          //Check whether fast sync was started before
+          if (fastSyncStateStorage.getSyncState().isDefined) {
```
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala]: Even if this is ok, it's confusing to me to retrieve if the fastSync has ever started doing `fastSyncStateStorage.getSyncState()` but to check if it has finished we need to do it based on  `appStateStorage.isFastSyncDone()`

Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala]: we may need new flag for this, but I would do it in separate PR
